configure.ac

```diff
@@ -19,35 +19,76 @@ AC_PROG_SED
 
 AC_CHECK_FUNCS(copy_file_range fgetxattr statx fgetpwent_r issetugid secure_getenv)
 
-AC_SEARCH_LIBS(cap_from_name, [cap], [AC_DEFINE([HAVE_CAP], 1, [Define if libcap is available])], [AC_MSG_ERROR([*** libcap headers not found])])
-AC_SEARCH_LIBS(seccomp_rule_add, [seccomp], [AC_DEFINE([HAVE_SECCOMP], 1, [Define if seccomp is available])], [AC_MSG_ERROR([*** libseccomp headers not found])])
-AC_SEARCH_LIBS(seccomp_arch_resolve_name, [seccomp], [AC_DEFINE([SECCOMP_ARCH_RESOLVE_NAME], 1, [Define if seccomp_arch_resolve_name is available])], [ ])
-AC_SEARCH_LIBS(sd_listen_fds, [systemd], [AC_DEFINE([HAVE_SYSTEMD], 1, [Define if libsystemd is available])], [ ])
+dnl libcap
+AC_ARG_ENABLE([caps],
+	AS_HELP_STRING([--disable-caps], [Ignore libcap and disable support]))
+AS_IF([test "x$enable_caps" != "xno"], [
+	AC_CHECK_HEADERS([sys/capability.h], [], [AC_MSG_ERROR([*** POSIX caps headers not found])])
+	AS_IF([test "$ac_cv_header_sys_capability_h" = "yes"], [
+		AC_SEARCH_LIBS(cap_from_name, [cap], [AC_DEFINE([HAVE_CAP], 1, [Define if libcap is available])], [AC_MSG_ERROR([*** libcap headers not found])])
+	])
+])
 
-AC_DEFINE([HAVE_SELINUX], 1, [Define if SELinux is available])
-AC_DEFINE([HAVE_APPARMOR], 1, [Define if AppArmor is available])
+dnl libseccomp
+AC_ARG_ENABLE([seccomp],
+	AS_HELP_STRING([--disable-seccomp], [Ignore libseccomp and disable support]))
+AS_IF([test "x$enable_seccomp" != "xno"], [
+	AC_CHECK_HEADERS([seccomp.h], [], [AC_MSG_ERROR([*** Missing libseccomp headers])])
+	AS_IF([test "$ac_cv_header_seccomp_h" = "yes"], [
+		AC_SEARCH_LIBS(seccomp_rule_add, [seccomp], [AC_DEFINE([HAVE_SECCOMP], 1, [Define if seccomp is available])], [AC_MSG_ERROR([*** libseccomp headers not found])])
+		AC_SEARCH_LIBS(seccomp_arch_resolve_name, [seccomp], [AC_DEFINE([SECCOMP_ARCH_RESOLVE_NAME], 1, [Define if seccomp_arch_resolve_name is available])], [ ])
+	])
+])
 
-AC_SEARCH_LIBS(yajl_tree_get, [yajl], [AC_DEFINE([HAVE_YAJL], 1, [Define if libyajl is available])], [AC_MSG_ERROR([*** libyajl headers not found])])
+dnl libsystemd
+AC_ARG_ENABLE([systemd],
+	AS_HELP_STRING([--enable-systemd], [Enable systemd support]))
+AS_IF([test "x$enable_systemd" = "xyes"], [
+	AC_CHECK_HEADERS([systemd/sd-bus.h], [], [AC_MSG_ERROR([*** Missing libsystemd headers])])
+	AS_IF([test "$ac_cv_header_systemd_sd_bus_h" = "yes"], [
+		AC_SEARCH_LIBS(sd_listen_fds, [systemd], [AC_DEFINE([HAVE_SYSTEMD], 1, [Define if libsystemd is available])], [AC_MSG_ERROR([*** Failed to find libsystemd])])
+	])
+])
+
+dnl libselinux
+AC_ARG_ENABLE([selinux],
+	AS_HELP_STRING([--enable-selinux], [Enable selinux support]))
+AS_IF([test "x$enable_selinux" = "xyes"], [
+	AC_DEFINE([HAVE_SELINUX], 1, [Define if SELinux is available])
+])
+
+dnl libapparmor
+AC_ARG_ENABLE([apparmor],
+	AS_HELP_STRING([--enable-apparmor], [Enable apparmor support]))
+AS_IF([test "x$enable_apparmor" = "xyes"], [
+	AC_DEFINE([HAVE_APPARMOR], 1, [Define if AppArmor is available])
+])
```

Comment:
we can probably drop these checks completely and always set `HAVE_SELINUX` and `HAVE_APPARMOR`.  We don't depend on `libselinux` and `libapparmor` in the code but we write directly to the `/proc` file system

Reply:
this should also solve the warnings we got from the compiler without requiring the `Making the pedantic preprocessor gods happy` patch

Reply:
I noticed, but perhaps it would make sense to keep it in, in case of library linking in the future. Otherwise I guess it can be dropped, it looks like kernel support is checked at runtime?

Reply:
yea, I'm having a passive aggressive day with GCC

Reply:
if we drop the optional enable, why not just drop the preprocessor macros as well?

Reply:
yes, I am all for it

Reply:
just removed in latest commit